src/User/Controller/SettingsController.php

```diff
@@ -488,4 +512,39 @@ protected function disconnectSocialNetwork($id)
         $account->delete();
         $this->trigger(SocialNetworkConnectEvent::EVENT_AFTER_DISCONNECT, $event);
     }
+
+    public function actionTwoFactorMobilePhone($id)
+    {
+        Yii::$app->response->format = Response::FORMAT_JSON;
+
+        /**
+        * 
+        *
+        * @var User $user 
+        */
+        $user = $this->userQuery->whereId($id)->one();
+
+        if (null === $user) {
+            return [
+                'success' => false,
+                'message' => Yii::t('usuario', 'User not found.')
+            ];
+        }
+        $mobilePhone = Yii::$app->request->get('mobilephone');
+        $currentMobilePhone = $user->getAuthTfMobilePhone();
+        $success=false;
+        if($currentMobilePhone==$mobilePhone){
+            $success=true;    
+        }else{
+            $success = $user->updateAttributes(['auth_tf_mobile_phone' => $mobilePhone]);
+            $this->make(TwoFactorSmsCodeGeneratorService::class, [$user])->run();
+        }
+        
+        return [
+                    'success' => $success,
+                    'message' => $success
+                    ? Yii::t('usuario', 'Mobile phone number successfully enabled.')
+                    : Yii::t('usuario', 'Mobile phone number not registered.'),
```

Comment:
What does the *not registered* message means? Maybe it would be more appropriate *Error while enabling SMS 2FA*?